### DISCUSSION TITLE
Invalidating Texture data

### DISCUSSION THREAD

**Discussion by barotto** [mikke89/RmlUi#233] [Q&A] [answered]
I have a window where there's a bunch of `<img>` elements inside a `<div>`, each with their `src` property pointing to a png file on the file system via and absolute path.

These png's will be updated externally and what I'm doing now to update the content of the `<div>` is calling `SetInnerRML("")` on it and rebuilding all the `<img>`'s.

For some of these png's the absolute file system path won't change, but their pixel data will. Unfortunately RmlUi won't request a `RenderInterface::LoadTexture()` again for an image that has been already loaded, so even if I rebuild an `<img>` its content will remain unchanged.

How can I tell RmlUi to invalidate the state of textures associated to `<img>` elements?

**Comment by mikke89** [comment, marked as answer]
This can be achieved by simply calling `Rml::ReleaseTextures()`. This way RmlUi will call `LoadTexture` again on all the textures.

https://github.com/mikke89/RmlUi/blob/7ef8ab965e6d597785c6854261042b5fea756c7c/Include/RmlUi/Core/Core.h#L153

**Reply by barotto** [reply]
So it's all or nothing? Bummer.

I'll peruse RmlUi's source and see if I can implement something for my use case, maybe a `ReleaseTexture(const String& source)` or something.

Thanks.

**Reply by mikke89** [reply]
No problem! I wouldn't mind a PR adding that functionality.
